backend/inspirehep/curation/views.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from flask import Blueprint, request
+from inspire_schemas.builders.literature import LiteratureBuilder
+from invenio_db import db
+
+from inspirehep.accounts.decorators import login_required_with_roles
+from inspirehep.accounts.roles import Roles
+from inspirehep.records.api import LiteratureRecord
+from inspirehep.serializers import jsonify
+
+from .utils import check_keyword_dict_datatypes
+
+blueprint = Blueprint("inspirehep_curation", __name__, url_prefix="/curation")
+
+
+@blueprint.route("/literature/<int:pid_value>/keywords", methods=["PUT"])
+@login_required_with_roles([Roles.cataloger.value])
+def add_keywords(pid_value):
+    json = request.json
+    required_fields = {"_desy_bookkeeping": dict, "keywords": list}
+
+    if not any(set(json) & set(required_fields)):
+        return (
+            jsonify(
+                success=False,
+                message=f"""{" or ".join(required_fields.keys())} must be passed""",
+            ),
+            400,
+        )
+
+    datatype_warnings = check_keyword_dict_datatypes(json, required_fields)
+    if datatype_warnings:
+        return (
+            jsonify(
+                success=False,
+                message=f"""{" ".join(datatype_warnings)}""",
+            ),
+            400,
+        )
```

Comment:
this is not something you should be doing manually but use `webargs` for this (validate the payload based on marshmallow schema).